src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -452,4 +448,42 @@ public static String getLastNameFromPath(String path) {
     String[] pathParts = StringUtils.split(path, '/');
     return pathParts[pathParts.length - 1];
   }
+
+  private void verifyAuthorization(String snapshotId, AuthenticatedUserRequest authUser) {
+    samAuthorizations.computeIfAbsent(
+        snapshotId,
+        id -> {
```

Comment:
This is wrong. It would mean that, if the snapshot has been authorized for a user and is in the cache, than ANY user could access the snapshot until it's evicted from the cache. Will fix.

Reply:
Good catch